tfhe/src/integer/ciphertext/boolean_value.rs

```diff
@@ -0,0 +1,105 @@
+use super::{IntegerCiphertext, IntegerRadixCiphertext};
+use crate::integer::{RadixCiphertext, ServerKey};
+use crate::shortint::Ciphertext;
+use serde::{Deserialize, Serialize};
+
+/// Wrapper type used to signal that the inner value encrypts 0 or 1
+///
+/// Since values ares encrypted, it is not possible to know whether a
+/// ciphertext encrypts a boolean value (0 or 1). However some algorithms
+/// require that the ciphertext does indeed encrypt a boolean value.
+///
+/// This wrapper serves as making it explicit that it is known that the value
+/// encrypted is 0 or 1. And that if a function taking a BooleanBlock as input
+/// returns incorrect value, it may be due to the value not really being 0 or 1.
+///
+/// Also some function such as comparisons are known to return an encrypted value
+/// that is either 0 or 1, and thus return a Ciphertext wrapped in a [BooleanBlock].
+///
+/// # Examples
+/// ## Converting a [BooleanBlock] to a [RadixCiphertext]
+///
+/// ```
+/// use tfhe::integer::{gen_keys_radix, BooleanBlock};
+/// use tfhe::shortint::parameters::PARAM_MESSAGE_2_CARRY_2_KS_PBS;
+///
+/// // We have 4 * 2 = 8 bits of message
+/// let size = 4;
+/// let (cks, sks) = gen_keys_radix(PARAM_MESSAGE_2_CARRY_2_KS_PBS, size);
+///
+/// let a = 128u8;
+/// let b = 55u8;
+///
+/// let ct_a = cks.encrypt(a);
+/// let ct_b = cks.encrypt(b);
+///
+/// let ct_res = sks.ge_parallelized(&ct_a, &ct_b);
+/// // Convert the boolean value to a RadixCiphertext of size blocks
+/// // so we can use it in operations
+/// let ct_res = ct_res.into_radix(size, &sks);
+/// // Decrypt:
+/// let dec: u8 = cks.decrypt(&ct_res);
+/// assert_eq!(u8::from(a >= b), dec);
+/// ```
+#[derive(Clone, Deserialize, Serialize)]
+pub struct BooleanBlock(pub(crate) Ciphertext);
+
+impl BooleanBlock {
+    /// Creates a new BooleanBlock without checks.
+    ///
+    /// You have to be sure the ciphertext encrypts 0 or 1 otherwise
+    /// functions expecting a BooleanBlock could result in wrong computation
+    pub fn new_unchecked(block: Ciphertext) -> Self {
+        Self(block)
+    }
+
+    /// Creates a new BooleanBlock, but does some checks to see
+    /// if it seems plausible that it encrypts a boolean.
+    ///
+    /// Sometimes these checks may prove to be too strict,
+    /// and you might need to use [BooleanBlock::new_unchecked] if you know it actually does encrypt
+    /// a boolean, or [BooleanBlock::convert] if there is some uncertainty.
+    pub fn try_new(block: Ciphertext) -> Option<Self> {
+        if block.degree.0 > 1 {
+            None
+        } else {
+            Some(Self(block))
+        }
+    }
```

Comment:
should we take a reference to an &IntegerCiphertext or IntegerRadixCiphertext and perform the checks for all blocks non only the LSB one ?

Reply:
made `trye_new`, `convert` take `&T: IntegerRadixCiphertext,`